Source/Fuse.Storage/iOS/IOSUserSettings.uno

```diff
@@ -0,0 +1,117 @@
+using Uno;
+using Uno.UX;
+using Uno.Compiler.ExportTargetInterop;
+
+namespace Fuse.Storage
+{
+	[ForeignInclude(Language.ObjC, "iOS/NSUserDefaults+SecureAdditions.h")]
+	extern(iOS) class IOSUserSettings : IUserSettings
+	{
+		ObjC.Object _handle;
+
+		public IOSUserSettings()
+		{
+			_handle = Init();
+		}
+
+		[Foreign(Language.ObjC)]
+		ObjC.Object Init()
+		@{
+			NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+			[userDefaults setSecret:@"UX1qMnfqmPm9GV1Bz6w5h1(41i80|x"];
```

Comment:
This module is implemented atop [NSUserDefaults](https://developer.apple.com/documentation/foundation/nsuserdefaults) on IOS and [`SharedPreferences`](https://developer.android.com/guide/topics/data/data-storage.html#pref) on Android. and they both persisted in binary and/or XML file format and stored in the app directory (sandboxed). 

In a perfect environment, this should be enough for security. But on Jailbroken device or the rooted device, they are easy to be found and be read. So in the `UserSettings` module, I have added extra layer security by encrypting every content that has been stored. Since `Uno` doesn't have cryptography modules, I have used third party code to do that [(here](https://github.com/tozny/java-aes-crypto/blob/master/aes-crypto/src/main/java/com/tozny/crypto/android/AesCbcWithIntegrity.java) for the java implementation and [here](https://github.com/kelp404/CocoaSecurity) for iOS. They are both have MIT license)


> What's this default secret?

Yes that is the default secret for implementation on IOS,  and [Here](https://github.com/fuse-open/fuselibs/pull/1304/files#diff-edee55c4ed6b676ef06911e9b418efa6R146) are default secret key for Android (it takes package name as password, and serial device number as a salt)

Reply:
So in jailbroken / rooted cases, how does the crypto keys or secrets get stored in a way that avoids inspection?

Also, I'm unsure if jailbroken and rooted devices require extra security. After all, the user explicitly broke the device security, so isn't this exactly what they asked for?

Reply:
Oh, I guess you already answered this, package name, seeded with device serial... Neither of those are secret, so how is this actual security and not security through obscurity?

Reply:
Sorry, I wasn't clear giving information about how the default secret key for Android generated, The actual mechanism is that first, the secret key used for encryption is randomly generated using a generator method that ship with third-party code [here](https://github.com/fuse-open/fuselibs/pull/1304/files#diff-e3eab82e75f16b753d8e9ed2d2e9bb38R136),  and then generated key is stored by a key name on the disk, the key name itself also generated by [this](https://github.com/fuse-open/fuselibs/pull/1304/files#diff-edee55c4ed6b676ef06911e9b418efa6R146) method where it takes package name and unique device id as default password and salt for generating key name.

But after all this effort, I think you were right it is meaningless on jailbroken/rooted device since there's no way to safely store crypto keys or secrets in a way that avoids inspection. 
Oh Uhm, any suggestion?

But if we step aside from the encryption function that exists in this module, I think the actual function of this module it's very useful for our users. 

So should I drop all encryption layer function in this module when storing and retrieving data?

Reply:
Usually passwords or secret keys would not be stored on the users device, only tokens. The encryption could work for a situation where you want to store a "secure note" type of data and you want to "password protect" it, the user could define the password which gets used as the secret and password, so no one could scan the device for the secret, in this scenario though, if the user loses their password, there's no way to get the data back, except for brute forcing it of course, although the user will probably pick something short and memorable so in the end I don't believe it will be effective without putting in other layers, like exponential time delay between wrong guesses.

Reply:
The encryption feature could probably become a module on its own and allow the developer to work that process out if they want it included for their user settings.

Reply:
@ichan-mb: I think it could at least make sense to introduce the functionality without the extra encryption as the first step. That seems very useful.

If people want to encrypt their storage for added security and have a good way of storing keys, we might provide them with some helpers for that. But I think that could come as a separate PR, and probably shouldn't be something that we facilitate for them. Which I guess is basically what @AndrewEQ suggests as well :)

Reply:
Ok then, I will drop extra encryption on this feature. Thank you all for the input :), I have learned a lot from this.